f3write.c

```diff
@@ -370,6 +370,12 @@ int main(int argc, char **argv)
 
 	restrict_fsaccess(&args.dev_path);
 
+#ifdef __OpenBSD__
+	assert(!unveil(args.dev_path, "rwc"));
+	assert(!unveil(NULL, NULL));
+	assert(!pledge("stdio rpath wpath cpath", ""));
+#endif
+
```

Comment:
Add a second `bool write` parameter to `restrict_fsaccess()` and move these lines to `restrict_fsaccess()`.